adagrams/game.py

```diff
@@ -1,11 +1,212 @@
+ 
+
+    for letter in word:
+        if letter in letter_bank_dict:
+            letter_bank_dict[letter] -= 1
+            if letter_bank_dict[letter] < 0:
+                return False
+        
+        else:
+            return False
+    return True
 
 def score_word(word):
-    pass
+    """
+    input: word
+    output: total score of the word: sum of letters' point value + bonus of lenth value 8 (if lenth >= 7)
+    score= 0
+    for letter in word:
+        for key, value in score_chart:
+            if letter in value:
+                score += key
+
+    if lenth > 7 and lenth < 11:
+        score += 8
+        
+scores_chart = {"A": 1, "C": 3, "B": 3, "E": 1, "D": 2, "G": 2, 
+         "F": 4, "I": 1, "H": 4, "K": 5, "J": 8, "M": 3, 
+         "L": 1, "O": 1, "N": 1, "Q": 10, "P": 3, "S": 1, 
+         "R": 1, "U": 1, "T": 1, "W": 4, "V": 4, "Y": 4, 
+         "X": 8, "Z": 10}
+    """
+    word = word.upper()
+    score = 0
+    for letter in word:
+        score += scores_chart[letter]
+    if len(word) >= 7 and len(word)< 11:
+        # if len(word) == 7 or len(word) == 8 or len(word) == 9 or len(word) == 10:
+        # if len(word) in [7, 8, 9, 10]:
+        score +=  8
 
+    return score 
+    
+
+    
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    """
+    input: word_list
+    output: (winning word, score)
+    tie-break rules: 10 letters > fewer letters  or  1st one if lenth is same
+    """
+
+    winning_words = []
+    score_dict = {}
+    winner = ""
+    for word in word_list:
+        score_dict[word] = score_word(word)
+    
+    max_score = max(score_dict.values())
+    for key, value in score_dict.items():
+        if value == max_score:
+            winning_words.append(key) 
+    # to get words with highest score include case with more than 1 highest score
+    
+    if len(winning_words) == 1:
+        winner = winning_words[0]
+    
+    else: 
+
+        for i in range(len(winning_words)):
+            if len(winning_words[i]) == 10:
+                winner = winning_words[i]
+                break
+            #when return the first word that has a lenth of 10 no matter how many lenth 10 words in the list
+            #words = ["MMMM", "WWW"]
+            #words = ["WWW", "MMMM"]
+            else:
+                min_length = len(winning_words[0])
+                i = 1
+                for i in range(len(winning_words)):
+                    if len(winning_words[i]) < min_length:
+                        min_length = len(winning_words[i]) 
+                        # to deside the min length in the winning_words list 
+                        winner = winning_words[i]  
+                    else:
+                        winner = winning_words[0]
+                    
+
+                
+            # return the first min length word no matter how many of them in the list
+    winner_list = [winner, score_word(winner)]
+    return tuple(winner_list)    
```

Comment:
No need to convert the list into a tuple.  python3 automatically returns a tuple when multiple values separated by a comma are in a return statement.

```python
return winner, score_word(winner)
```